tests/test_readwrite.py

```diff
@@ -368,42 +360,60 @@ def check_compressed(key, value):
     assert_equal(adata, expected)
 
 
-def test_zarr_compression(tmp_path):
-    from numcodecs import Blosc
-
-    pth = str(Path(tmp_path) / "adata.zarr")
-    adata = gen_adata((10, 8))
-    compressor = Blosc(cname="zstd", clevel=3, shuffle=Blosc.BITSHUFFLE)
-    not_compressed = []
-
-    ad.io.write_zarr(pth, adata, compressor=compressor)
-
-    def check_compressed(value, key):
-        if isinstance(value, ZarrArray):
-            if value.shape != ():
-                (read_compressor,) = value.compressors
-                print(read_compressor, key)
-                if read_compressor != compressor:
-                    not_compressed.append(key)
-
-    if is_zarr_v2():
-        with zarr.open(str(pth), "r") as f:
-            f.visititems(check_compressed)
-    else:
-        f = zarr.open(str(pth), mode="r")
-        for key, value in f.members(max_depth=None):
-            check_compressed(value, key)
+@pytest.mark.parametrize("zarr_write_format", [2, 3])
+def test_zarr_compression(tmp_path, zarr_write_format):
+    with ad.settings.override(zarr_write_format=zarr_write_format):
+        pth = str(Path(tmp_path) / "adata.zarr")
+        adata = gen_adata((10, 8))
+        if zarr_write_format == 2 or is_zarr_v2():
+            from numcodecs import Blosc
 
-    if not_compressed:
-        sep = "\n\t"
-        msg = (
-            f"These elements were not compressed correctly:{sep}"
-            f"{sep.join(not_compressed)}"
-        )
-        raise AssertionError(msg)
+            compressor = Blosc(cname="zstd", clevel=3, shuffle=Blosc.BITSHUFFLE)
+        else:
+            from zarr.codecs import BloscCodec
+
+            compressor = BloscCodec(cname="zstd", clevel=3, shuffle="bitshuffle")
+        not_compressed = []
+
+        ad.io.write_zarr(pth, adata, compressor=compressor)
+
+        def check_compressed(value, key):
```

Comment:
use some early returns here, I can’t see the code all the way over there in China! 😉